acceptance/builder_test.go

```diff
@@ -0,0 +1,418 @@
+				ctrPath("/layers"),
+				builderImage,
+				h.WithFlags(
+					"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+					"--env", "CNB_GROUP_PATH=/cnb/group_tomls/always_detect_group.toml",
+					"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+				),
+			)
+			// check builder metadata.toml for success test
+			md := getBuilderMetadata(t, filepath.Join(copyDir, "layers/config/metadata.toml"))
+
+			h.AssertStringContains(t, md.Buildpacks[0].API, "0.2")
+			h.AssertStringContains(t, md.Buildpacks[0].ID, "hello_world")
+			h.AssertStringContains(t, md.Buildpacks[0].Version, "0.0.1")
+		})
+	})
+
+	when("invalid input files", func() {
+		// .../cmd/lifecycle/builder.go#readData
+		when("group.toml", func() {
+			when("not found", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to read buildpack group"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+			//TODO: check some output file for this case not except any error message
+			when("empty", func() {
+				it("succeeds", func() {
+					h.DockerRunAndCopy(t,
+						containerName,
+						copyDir,
+						ctrPath("/layers"),
+						builderImage,
+						h.WithFlags(
+							"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+							"--env", "CNB_GROUP_PATH=/cnb/group_tomls/empty_group.toml",
+							"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						),
+					)
+					// check builder metadata.toml for success test
+					assertBuilderMetadata(t, filepath.Join(copyDir, "layers/config/metadata.toml"))
+				})
+			})
+
+			when("invalid", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/wrong_group.toml",
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to read buildpack group: Near line"
+					h.AssertStringContains(t, string(output), expected)
+				})
+			})
+
+			// .../cmd/lifecycle/builder.go#Exec
+			when("invalid builpack api", func() {
+				it("errors", func() {
+					command := exec.Command(
+						"docker",
+						"run",
+						"--rm",
+						"--env", "CNB_PLATFORM_API="+latestPlatformAPI,
+						"--env", "CNB_GROUP_PATH=/cnb/group_tomls/invalid_buildpack_api_group.toml",
+						"--env", "CNB_PLAN_PATH=/cnb/plan_tomls/always_detect_plan.toml",
+						builderImage,
+					)
+					output, err := command.CombinedOutput()
+					h.AssertNotNil(t, err)
+					expected := "failed to : parse buildpack API"
```

Comment:
This following error comes from https://github.com/buildpacks/lifecycle/blob/3297dacad3f0d58bcbbb44dff0d896b9ad1c28c1/cmd/version.go#L70 right?
Maybe it's worth changing the expected to the full error.

Reply:
fixed